### ISSUE TITLE
Items not stacking due to nbt mismatch

### ISSUE THREAD

**Issue by kkbizz** [NillerMedDild/Enigmatica6#97]
**Modpack version**
0.2.5

**Issue**
Sometimes items are not stacking because there is an nbt mismatch between them. 
I have an example with some glass i have in my inventory.
In the attached screenshot you see a random tag added to the 12 glass. 
I don't know what is causing this but i have seen it with sand as well.

**Logs**
![image](https://user-images.githubusercontent.com/4943652/89670352-5dbb0400-d8e1-11ea-816a-0921b7ec546c.png)

**Comment by kkbizz** [comment]
Spoke to soon, i think i figured out what causes this: inventory sorting. When sorting the inventory an extra tag gets added.

**Comment by NillerMedDild** [comment]
Thank you for the report, I will investigate the issue.

**Comment by mrDLSable** [comment]
I have had this issue as well.

**Comment by NillerMedDild** [comment]
I can't seem to reproduce this, how are you sorting?

**Comment by kkbizz** [comment]
middle mouse button om my inventory.
Info about the glass(maybe helpfull): smelted in vanilla furnace and the stack that was changed has been in my hotbar while placing.

**Comment by NillerMedDild** [comment]
Still nope, can't reproduce this :s

**Comment by Darkere** [comment]
What gui did you have open where sorting produces this issue?

**Comment by kkbizz** [comment]
just my inventory, i'll remove the "tagged" stack of items from my inventory and will be trying to find the source of the issue.

**Comment by kkbizz** [comment]
i did some more testing and cannot find the exact cause of this, it happens randomly.

**Comment by NillerMedDild** [comment]
Fair enough, I'll keep trying :)

**Comment by GrownUpBaby** [comment]
I have had this happen as well. I have noticed it with cobblestone and a few other blocks

**Comment by Mechalith** [comment]
I can tell you exactly what causes this, although I don't know why. FTB Ultimine causes anything it mined and dropped to not stack with items that weren't veinmined. This can be corrected by dropping the stacks that won't merge and picking them up simultaneously, but I haven't found a way to prevent the problem from happening.

**Comment by NillerMedDild** [comment]
I can't reproduce that Mechalith :/

**Comment by Mechalith** [comment]
Weird, it's happened every time I've checked it. I can get it to happen pretty consistently so I'll do a bit more testing and see if I can reliably break it.

**Comment by Mechalith** [comment]
I don't know if it's directly related or not, but I have noticed since the most recent update that I'm finding both the vanilla and Enigmatica version of a number of ores at times. I can convert them, but I was thinking that if there's more than one type of ore dropping then there might be multiple mostly-identical duplicate items or blocks with small variations in NBT data depending on which mod generated them. (I'm pretty sure all the vanilla ores I've seen have been in or around the cave campsites. Not sure if Better Caves is responsible for those generating or if it's Quark or Valhelsia Structures or something.)

Also, ever since I had my inaccurate epiphany about Ultimine, I haven't been able to reproduce the issue at all, regardless of what I've tried. Don't know whether that's due to the update or not.
